### PR TITLE
#22047 Feature: Newrelic transaction name based on CLI name

### PR DESCRIPTION
**Pull request by lbajsarowicz**
### Description (*)
Screenshot 2019-03-30 at 13 51 23

### Fixed Issues (if relevant)

1. magento/magento2#22047: Magento CRON Job Names are missing in NewRelic: "Transaction Names"

### Manual testing scenarios (*)
1. Configure your Newrelic setup
2. Run any CLI command
3. Check transaction list

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/NewRelicReporting/Model/NewRelicWrapper.php [hunk 1]
```diff
@@ -49,11 +49,24 @@ public function reportError($exception)
...
         }
     }
 
+    /**
+     * Wrapper for 'newrelic_name_transaction'
+     *
+     * @param string $transactionName
+     * @return void
+     */
+    public function setTransactionName(string $transactionName): void
```
Comment (sivaschenko) [inline app/code/Magento/NewRelicReporting/Model/NewRelicWrapper.php:63]: Adding a public function is a backward incompatible change, however, this class is not public (not marked with `@api` tag)

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-4634](https://jira.corp.magento.com/browse/ENGCOM-4634) has been created to process this Pull Request

**Comment by meker12** [comment]
Release note for version 2.3.x:

New Relic Reporting:

Magento 2 CLI command names are now reported in the transaction names in the New Relic APM Monitoring Data. The transaction name is based on the Magento CLI command name, for example `cli app:config:import`, `cli indexer:reindex`,  and `cli cache:flush`.   In previous releases, the New Relic transaction names for CLI commands initiated by Magento 2 cron jobs was missing and reported as `unknown`.

**Comment by lbajsarowicz** [comment]
@meker12 Not "Magento 2 cron job names" but "Magento 2 CLI command names" to be precise (!)

**Comment by meker12** [comment]
@lbajsarowicz  Thanks so much for the feedback 🥇

**Pull request merged**